src/main/java/com/mixpanel/android/viewcrawler/ImageStore.java

```diff
@@ -0,0 +1,138 @@
+package com.mixpanel.android.viewcrawler;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.util.Base64;
+import android.util.Log;
+
+import com.mixpanel.android.util.HttpService;
+import com.mixpanel.android.util.RemoteService;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * ABSOLUTELY NOT thread, or even process safe.
+ * Writes and reads files and directories at known paths, and uses a shared instance of MessageDigest.
+ */
+/* package */ class ImageStore {
+    public ImageStore(Context context) {
+        this(
+            context.getDir(DEFAULT_DIRECTORY_NAME, Context.MODE_PRIVATE),
```

Comment:
Do we need the `Context.MODE_PRIVATE` here? Here's the comment from Context.java
```
    /**
     * File creation mode: the default mode, where the created file can only
     * be accessed by the calling application (or all applications sharing the
     * same user ID).
     * @see #MODE_WORLD_READABLE
     * @see #MODE_WORLD_WRITEABLE
     */
    public static final int MODE_PRIVATE = 0x0000;
```